frontend/src/components/Header.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import MonsterContext from '../contexts/MonsterContext';
+import styled from 'styled-components/macro';
+import { useContext } from 'react';
+import { useHistory } from 'react-router-dom';
+import { BsStar, BsCheck, BsFillPlusCircleFill } from 'react-icons/bs';
+
+export default function Header({
+  icons,
+  currentMonsterId,
+  task,
+  monster,
+  add,
+}) {
+  const { monsters } = useContext(MonsterContext);
+  const history = useHistory();
+
+  const monstersToDisplay = monsters?.filter(
+    (monster) => monster.id !== currentMonsterId
+  );
+
+  const LinkTasksOrRewards = (task, monsterId) => {
```

Comment:
Could you instead pass a string to the component? This could maybe help to make your component more stupid.

Reply:
yes, that would also make it more reader-friendly, but wouldn't that also lead to more potential errors?